index.rst

```diff
@@ -629,7 +629,67 @@ When you connect to the computer running a CSC and edit the configuration direct
   version: heads/tickets/DM-12345-0-g79e2257-dirty
 
 When this happens, it prevents us from precisely identifying what configuration was used.
-In this case, the preferred solution is to use :ref:`section-on-the-fly-config` to ensure traceability is not lost, at the expense of a couple extra minutes.
+In this case, the preferred solution is to follow the `On-the-Fly Configuration Changes`_ process to ensure traceability is not lost, at the expense of a couple extra minutes.
+
+.. _section-configuration-interaction-traceability:
+
+Finding and Using a Previously Used Configuration
+=================================================
+
+In the future, one may want to verify which configuration was being used for a given observation and possibly load the exact same configuration.
+Because we often use generic filenames (e.g. `simple_algorithm.yaml`), and file contents can change with time, creating a robust version controlled system must go beyond simply changing filenames.
+For this reason, additional metadata is associated with each configuration, notably the ``url`` and ``version`` parameters in both the ``configurationsAvailable`` and ``configurationApplied`` events.
```

Comment:
I suggest we publish the configuration used, as a json string. I realize that would be a big change, since it requires a new generic topic, which in turn requires another RFC or similar. I am not suggesting holding up this document for it.

Reply:
I'm not against this, but Michael should chime in. I'm not sure it's feasible to do this for all components, but maybe we can recommend a standard? 
As you say, this is beyond the scope of this document though.